x-pack/plugins/dashboard_enhanced/public/services/drilldowns/dashboard_to_dashboard_drilldown/drilldown.tsx

```diff
@@ -0,0 +1,163 @@
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { CoreStart } from 'src/core/public';
+import { reactToUiComponent } from '../../../../../../../src/plugins/kibana_react/public';
+import { SharePluginStart } from '../../../../../../../src/plugins/share/public';
+import { DASHBOARD_APP_URL_GENERATOR } from '../../../../../../../src/plugins/dashboard/public';
+import { PlaceContext, ActionContext, Config } from './types';
+import { CollectConfigContainer } from './components';
+import { DASHBOARD_TO_DASHBOARD_DRILLDOWN } from './constants';
+import { DrilldownDefinition as Drilldown } from '../../../../../drilldowns/public';
+import { txtGoToDashboard } from './i18n';
+import { DataPublicPluginStart, esFilters } from '../../../../../../../src/plugins/data/public';
+import { VisualizeEmbeddableContract } from '../../../../../../../src/plugins/visualizations/public';
+import {
+  isRangeSelectTriggerContext,
+  isValueClickTriggerContext,
+} from '../../../../../../../src/plugins/embeddable/public';
+
+export interface Params {
+  getSavedObjectsClient: () => CoreStart['savedObjects']['client'];
+  getApplicationService: () => Pick<CoreStart['application'], 'getUrlForApp' | 'navigateToApp'>;
+  getGetUrlGenerator: () => SharePluginStart['urlGenerators']['getUrlGenerator'];
+  getDataPluginActions: () => DataPublicPluginStart['actions'];
+}
+
+export class DashboardToDashboardDrilldown
+  implements Drilldown<Config, PlaceContext, ActionContext<VisualizeEmbeddableContract>> {
+  constructor(protected readonly params: Params) {
+    this.getDestinationUrl = this.getDestinationUrl.bind(this);
+  }
+
+  public readonly id = DASHBOARD_TO_DASHBOARD_DRILLDOWN;
+
+  public readonly order = 100;
+
+  public readonly getDisplayName = () => txtGoToDashboard;
+
+  public readonly euiIcon = 'dashboardApp';
+
+  private readonly ReactCollectConfig: React.FC<CollectConfigContainer['props']> = props => (
+    <CollectConfigContainer {...props} deps={this.params} />
+  );
+
+  public readonly CollectConfig = reactToUiComponent(this.ReactCollectConfig);
+
+  public readonly createConfig = () => ({
+    dashboardId: '',
+    useCurrentFilters: true,
+    useCurrentDateRange: true,
+  });
+
+  public readonly isConfigValid = (config: Config): config is Config => {
+    if (!config.dashboardId) return false;
+    return true;
+  };
+
+  public readonly getHref = async (
+    config: Config,
+    context: ActionContext<VisualizeEmbeddableContract>
+  ): Promise<string> => {
+    const { getUrlForApp } = await this.params.getApplicationService();
+    const dashboardPath = await this.getDestinationUrl(config, context);
+    // note: extracting hash and using 'kibana' as appId will be redundant,
+    // when dashboard move to np urls. (urlGenerator generates np url, which is not supported yet)
+    const dashboardHash = dashboardPath.split('#')[1];
+    return getUrlForApp('kibana', {
```

Comment:
Shall we extract `'kibana'` string from the `dashboardPath` created by URL generator?

Reply:
fyi, url generator generates newer url format. Instead of `kibana` in the path, there is `dashboard`. this will very soon be the correct url for dashboard app.

Reply:
> Instead of `kibana` in the path, there is `dashboard`.

I didn't know that. Now I'm thinking, shoud we change url generator to return `kibana`? And maybe that `kibana` could be a constant in code, which later can be changed to `dashboard`.